Jenkinsfile

```diff
@@ -299,55 +306,73 @@ pipeline {
+                        runTest('auto-config', 'cluster1')
+                        runTest('config', 'cluster1')
+                        ShutdownCluster('cluster1')
+                    }
                 }
-            }
-            steps {
-                CreateCluster('basic')
-                runTest('auto-config', 'basic')
-                runTest('config', 'basic')
-                runTest('init-deploy', 'basic')
-                runTest('monitoring', 'basic')
-                runTest('semi-sync', 'basic')
-                runTest('service-per-pod', 'basic')
-                runTest('scaling', 'basic')
-                runTest('sidecars', 'basic')
-                runTest('users', 'basic')
-                runTest('limits', 'basic')
-                ShutdownCluster('basic')
-            }
-        }
-        stage('E2E Backup Tests') {
-            when {
-                expression {
-                    !skipBranchBuilds
+                stage('E2E Cluster2') {
+                    when {
+                        expression {
+                            !skipBranchBuilds
+                        }
+                    }
+                    agent {
+                        label 'docker'
+                    }
+                    steps {
+                        prepareNode()
+                        unstash "sourceFILES"
+                        CreateCluster('cluster2')
+                        runTest('demand-backup', 'cluster2')
+                        runTest('scaling', 'cluster2')
+                        runTest('users', 'cluster2')
+                        ShutdownCluster('cluster2')
+                    }
                 }
-            }
-            steps {
-                CreateCluster('backup')
-                runTest('demand-backup', 'backup')
-                ShutdownCluster('backup')
-            }
-        }
-        stage('E2E GR Tests') {
-            when {
-                expression {
-                    !skipBranchBuilds
+                stage('E2E Cluster3') {
+                    when {
+                        expression {
+                            !skipBranchBuilds
+                        }
+                    }
+                    agent {
+                        label 'docker'
+                    }
+                    steps {
+                        prepareNode()
+                        unstash "sourceFILES"
+                        CreateCluster('cluster3')
+                        runTest('init-deploy', 'cluster3')
+                        runTest('gr-init-deploy', 'cluster3')
+                        runTest('monitoring', 'cluster3')
+                        runTest('semi-sync', 'cluster3')
+                        runTest('service-per-pod', 'cluster3')
+                        runTest('sidecars', 'cluster3')
+                        runTest('limits', 'cluster3')
+                        ShutdownCluster('cluster3')
+                    }
```

Comment:
@tplavcic how did you distribute tests via clusters?

Reply:
I had a test run and put test timings into google sheet and then split them into 3 clusters to have approximately the same runs in each cluster.

Reply:
nice)) I did the same!